### PR TITLE
added shopping cart and cart items enities

### PR DESCRIPTION
**Pull request by SaiyaJ1N**

### CODE REVIEW

**Review by ihorvasylenko23 — APPROVED** [review]
Nice job) Don`t forget about emty lines at the end of the files

File: src/main/java/com/onlinebookshop/service/impl/ShoppingCartServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final BookService bookService;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public ShoppingCartDto getShoppingCart() {
+        ShoppingCartDto shoppingCartDto = shoppingCartMapper.toDto(getCartOfCurrentUser());
+        shoppingCartDto.setCartItems(getCartOfCurrentUser().getCartItems().stream()
```
Comment (vovasalyha) [inline src/main/java/com/onlinebookshop/service/impl/ShoppingCartServiceImpl.java]: why are you calling `getCartOfCurrentUser()` 2 times? Considering that you method is not marked Transactional it's 2 queries to the DB

File: src/main/java/com/onlinebookshop/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                .map(cartItemMapper::toResponseDto)
+                .collect(Collectors.toSet()));
+        return shoppingCartDto;
+    }
+
+    @Override
```
Comment (vovasalyha) [inline src/main/java/com/onlinebookshop/service/impl/ShoppingCartServiceImpl.java:45]:
```suggestion
    @Override
    @Transactional
```

File: src/main/java/com/onlinebookshop/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        cartItem.setBook(bookService.findBookById(requestDto.getBookId()));
+        cartItem.setQuantity(requestDto.getQuantity());
+        CartItemResponseDto cartItemDto = cartItemMapper
+                .toResponseDto(cartItemRepository.save(cartItem));
+        ShoppingCartDto shoppingCartDto = shoppingCartMapper.toDto(cart);
+        shoppingCartDto.getCartItems().add(cartItemDto);
+        return cartItemDto;
+    }
+
+    @Override
```
Comment (vovasalyha) [inline src/main/java/com/onlinebookshop/service/impl/ShoppingCartServiceImpl.java:60]:
```suggestion
    @Override
    @Transactional
```

File: src/main/java/com/onlinebookshop/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    public void createShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setId(user.getId());
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getCartOfCurrentUser() {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication()
```
Comment (vovasalyha) [inline src/main/java/com/onlinebookshop/service/impl/ShoppingCartServiceImpl.java]: I would move fetching of auth user to user related service

File: src/main/resources/db/changelog/changes/08-create-cart-item-table.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            tableName: cart_item
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/08-create-cart-item-table.yaml:16]: don't forget about foreign keys for references to keep your DB consistent

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ. Let's improve it a bit

File: src/main/java/com/onlinebookshop/service/impl/ShoppingCartServiceImpl.java (hunk 1 above)
Comment (andrii-hoienko) [reply src/main/java/com/onlinebookshop/service/impl/ShoppingCartServiceImpl.java]: not resolved

File: src/main/resources/db/changelog/changes/11-insert-roles-and-admin-user.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      changes:
+        - insert:
+            tableName: user_role
+            columns:
+              - column:
+                  name: user_id
+                  valueComputed: (SELECT id FROM users WHERE email='admin@example.com')
+              - column:
+                  name: role_id
+                  valueComputed: (SELECT id FROM roles WHERE name='ROLE_ADMIN')
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/11-insert-roles-and-admin-user.yaml]: please add an empty lines

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Well done! But one comments is still not resolved

**Review by andrii-hoienko — APPROVED** [review]
GJ!

**Pull request merged**